core/SiteContentDetector.php

```diff
@@ -163,97 +191,104 @@ public function detectContent(array $detectContent = [SiteContentDetector::ALL_C
 
         // A request was made to get this data and it isn't currently cached, so write it to the cache now
         $cacheLife = (60 * 60 * 24 * 7);
-        $this->savePropertiesToCache($cacheKey, $requiredProperties, $cacheLife);
+        $this->saveToCache($cacheKey, $cacheLife);
     }
 
     /**
-     * Returns an array of properties required by the detect content array
+     * Returns if the detection with the provided id was detected or not
      *
-     * @param array $detectContent
+     * Note: self::detectContent needs to be called before.
      *
-     * @return array
+     * @param string $detectionClassId
+     * @return bool
      */
-    private function getRequiredProperties(array $detectContent): array
+    public function wasDetected(string $detectionClassId): bool
     {
-        $requiredProperties = [];
-
-        if (in_array(SiteContentDetector::CONSENT_MANAGER, $detectContent) || in_array(SiteContentDetector::ALL_CONTENT, $detectContent)) {
-            $requiredProperties = array_merge($requiredProperties, ['consentManagerId', 'consentManagerName', 'consentManagerUrl', 'isConnected']);
-        }
-
-        if (in_array(SiteContentDetector::GA3, $detectContent) || in_array(SiteContentDetector::ALL_CONTENT, $detectContent)) {
-            $requiredProperties[] = 'ga3';
-        }
-
-        if (in_array(SiteContentDetector::GA4, $detectContent) || in_array(SiteContentDetector::ALL_CONTENT, $detectContent)) {
-            $requiredProperties[] = 'ga4';
-        }
-
-        if (in_array(SiteContentDetector::GTM, $detectContent) || in_array(SiteContentDetector::ALL_CONTENT, $detectContent)) {
-            $requiredProperties[] = 'gtm';
-        }
-
-        if (in_array(SiteContentDetector::CMS, $detectContent) || in_array(SiteContentDetector::ALL_CONTENT, $detectContent)) {
-            $requiredProperties[] = 'cms';
-        }
-
-        if (in_array(SiteContentDetector::JS_FRAMEWORK, $detectContent) || in_array(SiteContentDetector::ALL_CONTENT, $detectContent)) {
-            $requiredProperties[] = 'jsFramework';
+        foreach ($this->detectedContent as $type => $detectedClassIds) {
+            if (array_key_exists($detectionClassId, $detectedClassIds)) {
+                return $detectedClassIds[$detectionClassId] ?? false;
+            }
         }
 
-        return $requiredProperties;
+        return false;
     }
 
     /**
-     * Checks that all required properties are in the cache array
-     *
-     * @param array $properties
-     * @param array $cache
+     * Returns an array containing ids of all detected detections of the given type
      *
-     * @return bool
+     * @param string $type One of the SiteContentDetectionAbstract::TYPE_* constants
+     * @return array
      */
-    private function checkCacheHasRequiredProperties(array $properties, array $cache): bool
+    public function getDetectsByType(string $type): array
```

Comment:
btw just noticed we're actually passing here integers and not string. Maybe not too important though
<img width="547" alt="image" src="https://github.com/matomo-org/matomo/assets/273120/86c6a629-2b76-4b0e-9bcf-593da355b01c">
just fyi @sgiehl

Reply:
Good catch. I'll update that in the PR for changing the layout